### PR TITLE
Supporting Light Private Ledger 

### PR DESCRIPTION
**Pull request by duanbing**
## Description

A plugin for Light private ledger,  enabling confidential computing in smart contrat, and data encryption in ledger.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Brief of your solution

refer to the wiki. 

## How Has This Been Tested?
1. start the off-chain confidential service;  
2. compile libmesateesdk.so.0.0.1 from https://github.com/xuperdata/teesdk/tree/master/xchain_plugin
2. add teeConfig  in xchain.conf 
```
teeConfig:
  enable: on
  tmsport: 5554
  uid: "uid1"
  token: "token1"
  pluginPath: "/var/lib/libmesateesdk.so.0.0.1"
  auditors:
  -
    publicder: /root/release/services/auditors/godzilla/godzilla.public.der
    sign: /root/release/services/auditors/godzilla/godzilla.sign.sha256
    enclaveinfoconfig: /root/release/services/enclave_info.toml
```
4. test by  [example](https://github.com/duanbing/xuper-sdk-go/blob/master/example/main_trust_counter.go)

### CODE REVIEW

File: core/contractsdk/cpp/example/trustops/src/trust_counter.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const std::string& creator = ctx->arg("creator");
+    if (creator.empty()) {
+        ctx->error("missing creator");
+        return;
+    }
+    ctx->put_object("creator", creator);
+    ctx->ok("initialize succeed");
+}
+
+DEFINE_METHOD(Counter, increase) {
```
Comment (fxsjy) [inline core/contractsdk/cpp/example/trustops/src/trust_counter.cc]: this method just implement store not increase. It is better to rename it to "store" or implement "increase".

File: core/contractsdk/cpp/src/xchain/trust_operators/trust_operators.cc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        return false;
+    }
+    auto ret = response->ParseFromString(std::string(out, out_size));
+    free(out);
+    return ret;
+}
+
+TrustOperators::TrustOperators(const std::string& addr):_address(addr){}
+
+bool TrustOperators::store(const uint32_t svn, const std::string& args,
```
Comment (icexin) [inline core/contractsdk/cpp/src/xchain/trust_operators/trust_operators.cc]: It is recommended to combine `put_object` to directly store data here, which is more in line with the semantics of `store`

File: core/contractsdk/cpp/example/trustops/src/trust_counter.cc (hunk 1 above)
Comment (duanbing) [reply core/contractsdk/cpp/example/trustops/src/trust_counter.cc]: fixed

File: core/common/config/config.go [hunk 3]
```diff
@@ -177,6 +177,25 @@ type WasmConfig struct {
 	EnableDebugLog bool
 	DebugLog       LogConfig
 	EnableUpgrade  bool
+	TEEConfig      *TEEConfig `yaml:"-"`
+}
+
+// copy from https://github.com/xuperdata/teesdk/blob/master/config.go to keep code clear
+type TEEConfig struct {
```
[the remark this answers is not in the document]
Comment (duanbing) [reply core/common/config/config.go]: commented

File: core/contractsdk/cpp/src/xchain/trust_operators/trust_operators.cc (hunk 2 above)
Comment (duanbing) [reply core/contractsdk/cpp/src/xchain/trust_operators/trust_operators.cc]: done.

**Pull request merged**